### PR TITLE
misc: Depend on 'cshim', if available, instead of host freestanding headers

### PR DESCRIPTION
**Pull request by mintsuki**

### CODE REVIEW

**Comment by Qwinci** [comment]
LGTM other than the cshim wrap is missing.

**Comment by mintsuki** [comment]
Yeah, it didn't apply because it is gitignored, oops. Fixed.

File: options/ansi/include/limits.h [hunk 1]
```diff
@@ -3,6 +3,10 @@
 
 #define CHAR_BIT 8
 
+#ifndef MB_LEN_MAX
+# define MB_LEN_MAX 1
```
Comment (Qwinci) [inline options/ansi/include/limits.h:7]: Shouldn't this be 4 for UTF-8? I am not sure if we actually implement any of the conversions functions properly yet but when we do this should likely be 4.

**Review by Qwinci — CHANGES_REQUESTED** [review]

File: options/ansi/include/limits.h (hunk 1 above)
Comment (mintsuki) [reply options/ansi/include/limits.h:7]: On Linux/glibc it appears to be 16... not sure.

Comment (Qwinci) [reply options/ansi/include/limits.h]: On musl it is 4, maybe it being bigger on glibc has something to do with locales.

**Review by qookei — APPROVED** [review]

**Review by no92 — APPROVED** [review]
LGTM, this builds managarm x86_64.

**Pull request merged**